subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNode.java

```diff
@@ -229,4 +234,22 @@ public void resolveMutations() {
             }
         }
     }
+
+    @Override
+    public boolean allDependenciesSuccessful() {
```

Comment:
I wonder what we would want to do for artifact transforms which run on the outputs of verification tasks to transform them into something which would be aggregated. Shouldn't those also respect verification failures?

So maybe the `isDirectDependency()` and this code should go into `Node` instead. And only `LocalTaskNode` overrides the `isDirectDependency()` with something which doesn't always return `false`.

Reply:
This is a good idea.

Reply:
This worked out well. I like the fluency of:
```
    public boolean allDependenciesSuccessful() {
        return dependencySuccessors.stream().allMatch(this::shouldContinueExecution);
    }
```

See also `shouldContinueExecution(Node)` and `isLifecycleDependency(Node)`.